edalize/libero.py

```diff
@@ -186,19 +194,39 @@ def tcl_file_filter(self, f):
             return file_types[_file_type] + f.name
         return ""
 
-    def constraint_file_filter(self, f, type="ALL"):
+    def syn_constraint_file_filter(self, f):
+        file_types = {
+            "SDC": "constraint/",
+            "FDC": "constraint/",
+            "NDC": "constraint/",
+        }
+        _file_type = f.file_type.split("-")[0]
+        if _file_type in file_types:
+            return f.name
+        return ""
+
+    def pnr_constraint_file_filter(self, f):
         file_types = {
             "PDC": "constraint/io/",
             "SDC": "constraint/",
             "FPPDC": "constraint/fp/",
         }
         _file_type = f.file_type.split("-")[0]
         if _file_type in file_types:
-            filename = f.name.split("/")[-1]
-            if type == "ALL":
-                return file_types[_file_type] + filename
-            elif _file_type == type:
-                return file_types[_file_type] + filename
+            return f.name
+        return ""
+
+    def constraint_file_filter(self, f, type="ALL"):
```

Comment:
Do we still need this function now that we have the syn and pnr versions?

Reply:
It was still used in one place for the timing constraints, but I replaced that with a simpler filter and removed this one.